boa/src/syntax/parser/statement/mod.rs

```diff
@@ -214,22 +214,24 @@ where
 ///
 /// [spec]: https://tc39.es/ecma262/#prod-StatementList
 #[derive(Debug, Clone, Copy)]
-pub(super) struct StatementList {
+pub(super) struct StatementList<'a> {
     allow_yield: AllowYield,
     allow_await: AllowAwait,
     allow_return: AllowReturn,
     break_when_closingbraces: bool,
     in_block: bool,
+    break_nodes: &'a [TokenKind],
 }
 
-impl StatementList {
+impl<'a> StatementList<'a> {
     /// Creates a new `StatementList` parser.
     pub(super) fn new<Y, A, R>(
         allow_yield: Y,
         allow_await: A,
         allow_return: R,
         break_when_closingbraces: bool,
         in_block: bool,
+        break_nodes: &'a [TokenKind],
```

Comment:
I think it makes sense to enforce this to be `&'static`, and that way we remove some extra lifetimes. I don't think it makes sense to customize this on runtime.

Reply:
This is purely due to my lack of understanding Rust lifetimes past what the compiler tells me to do :sweat_smile: I'll switch it up :100: